dateparser/date_parser.py

```diff
@@ -32,10 +33,103 @@ def __add__(self, other):
 parser.relativedelta.relativedelta = new_relativedelta
 
 
+class new_parser(parser.parser):
+    """
+    Implements an alternate parse method which supports preference to dates in future and past.
+    For more see issue #36
+    """
+
+    def parse(self, timestr, default=None, ignoretz=False, **kwargs):
+        # timestr needs to be a buffer as required by _parse
+        if isinstance(timestr, binary_type):
+            timestr = timestr.decode()
+
+        # Parse timestr
+        res = self._parse(timestr, **kwargs)
+
+        if res is None:
+            raise ValueError("unknown string format")
+
+        # Fill in missing date
+        new_date, given_fields = new_parser._populate(res, default)
+
+        # Correct date to if prefer in future of past
+        new_date = new_parser._correct(new_date, given_fields, default)
+
+        # Clean hour and minutes, etc in case not defined
+        if not res.hour and not res.minute and not res.second and not res.microsecond:
+            new_date = new_date.replace(hour=0, minute=0, second=0, microsecond=0)
+
+        return new_date
+
+    @staticmethod
+    def _populate(res, default):
+        given_fields = []  # fields that cannot be overriden
+        new_date = default
+
+        # Populate all fields
+        repl = {}
+        for field in ['year', 'month', 'day', 'hour', 'minute', 'second', 'microsecond']:
+            value = getattr(res, field)
+            if value is not None:
+                repl[field] = value
+        new_date = new_date.replace(**repl)
+
+        # Fix weekday
+        if res.weekday and not res.day:
+            new_date = new_date + new_relativedelta(weekday=res.weekday)
+
+        return new_date, [key + 's' for key in repl.keys()]
```

Comment:
Just use `return self._correct(new_date, repl)` here instead of returning a tuple.